src/Domain/Services/CreateAccount.php

```diff
@@ -139,14 +139,27 @@ protected function should_create_customer_account( \WP_REST_Request $request ) {
 		}
 
 		// From here we know that the shopper is not logged in.
+		// check for whether account creation is enabled at the global level.
+		$checkout = WC()->checkout();
 
-		if ( false === filter_var( get_option( 'woocommerce_enable_guest_checkout' ), FILTER_VALIDATE_BOOLEAN ) ) {
+		if ( $checkout instanceof \WC_Checkout && false === filter_var(
+			$checkout->is_registration_required(),
+			FILTER_VALIDATE_BOOLEAN
+		) ) {
 			// Store requires an account for all checkouts (purchases).
 			// Create an account independent of shopper option in $request.
 			// Note - checkbox is not displayed to shopper in this case.
 			return true;
 		}
 
+		if ( $checkout instanceof \WC_Checkout && false === filter_var(
```

Comment:
Should we return early if `! $checkout instanceof \WC_Checkout` instead of multiple checks?

Reply:
Yes we can 4cdb22e4ff8809e91897b97ce2ac697c7840e549